src/controllers/signAccountOp/signAccountOp.ts

```diff
@@ -1223,6 +1251,11 @@ export class SignAccountOpController extends EventEmitter {
           }
         }
 
+        // query the application state from memory to understand if the user
+        // hasn't actually rejected the request while waiting for the
+        // paymaster to respond
+        if (!this.#isSignRequestStillActive(this.accountOp)) return
+
```

Comment:
Note: This is personal opinion:
I don't really like this approach of passing functions down to sub-controllers which check for data of other controllers. Imo this breaks our one-way flow, although it is used in many controllers. Instead, I think you can add a boolean attribute to the controller `isSignRequestStillActive` and update it from main when the account op is rejected- main updates a sub-controller instead of sub-controller calls a function of main.

Reply:
I don't think it works this way. I tried changing the sign account op state while it was waiting for the paymaster response and it didn't work because on reject, we make signAccountOp null. But signAccountOp remains in memory and it just continues on with its state. That's why I tried querying the application state in this scenario

Reply:
well, change the state first and then destroy the account op

Reply:
Refactor however possible